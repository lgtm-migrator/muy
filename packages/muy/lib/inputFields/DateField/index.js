import React from "react"
import PropTypes from "prop-types"
import TextField from "@material-ui/core/TextField"

const DateField = ({
  id,
  inputProps,
  InputLabelProps,
  label,
  max,
  min,
  name,
  placeholder,
  required,
}) => (
  <TextField
    id={id}
    label={label}
    InputLabelProps={{
      shrink: true,
      ...InputLabelProps,
    }}
    inputProps={{
      "aria-required": required,
      autoCapitalize: "none",
      autoCorrect: "off",
      "data-name": name,
      max,
      min,
      size: 10,
      ...inputProps,
    }}
    name={name}
    placeholder={placeholder}
    required={required}
    type={"date"}
  />
)

export default DateField

DateField.defaultProps = {
  id: "date",
  name: "date",
}

DateField.propTypes = {
  id: PropTypes.string,
  InputLabelProps: PropTypes.object,
  inputProps: PropTypes.object,
  label: PropTypes.string,
  max: PropTypes.string,
  min: PropTypes.string,
  name: PropTypes.string,
  placeholder: PropTypes.string,
  required: PropTypes.bool,
}
