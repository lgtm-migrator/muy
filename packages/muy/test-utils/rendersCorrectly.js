import renderer from "react-test-renderer"

const rendersCorrectly = component => {
  const tree = renderer.create(component).toJSON()
  return expect(tree).toMatchSnapshot()
}

export default rendersCorrectly
