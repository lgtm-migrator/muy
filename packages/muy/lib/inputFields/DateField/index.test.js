import React from "react"
import DateField from "."
import rendersCorrectly from "../../../test-utils/rendersCorrectly"

describe("<DateField /> component", () => {
  it("renders correctly with no props", () => {
    rendersCorrectly(<DateField />)
  })
  it("renders correctly with label prop", () => {
    rendersCorrectly(<DateField label={"Date Label Text"} />)
  })
})
