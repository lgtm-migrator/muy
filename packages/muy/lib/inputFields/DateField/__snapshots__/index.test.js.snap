// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateField /> component should render correctly with 'outlined' variant 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoCapitalize="none"
      autoCorrect="off"
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input"
      data-name="date"
      disabled={false}
      id="date"
      name="date"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      size={10}
      type="date"
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-2"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`<DateField /> component should render correctly with label prop 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
    data-shrink={true}
    htmlFor="date"
    id="date-label"
  >
    Date Label Text
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoCapitalize="none"
      autoCorrect="off"
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      data-name="date"
      disabled={false}
      id="date"
      name="date"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      size={10}
      type="date"
    />
  </div>
</div>
`;

exports[`<DateField /> component should render correctly with no props 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoCapitalize="none"
      autoCorrect="off"
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      data-name="date"
      disabled={false}
      id="date"
      name="date"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      size={10}
      type="date"
    />
  </div>
</div>
`;
